### PR TITLE
DOC: update the api.types.is_number docstring

### PR DESCRIPTION
**Pull request by tapumar**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [X] PR title is "DOC: update the <your-function-or-method> docstring"
- [X] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [X] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [X] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
Docstring for "pandas.core.dtypes.inference.is_number" correct. :)
```

### CODE REVIEW

File: pandas/core/dtypes/inference.py [hunk 1]
```diff
@@ -28,20 +28,33 @@ def is_number(obj):
     """
     Check if the object is a number.
 
+    Booleans are valid because they are int subclass.
```
Comment (jreback) [inline pandas/core/dtypes/inference.py]: can you put this in Notes

File: pandas/core/dtypes/inference.py [hunk 2]
```diff
@@ -28,20 +28,33 @@ def is_number(obj):
...
+
     Examples
     --------
-    >>> is_number(1)
+    >>> pd.api.types.is_number(1)
+    True
+    >>> pd.api.types.is_number(7.15)
     True
-    >>> is_number("foo")
+    >>> pd.api.types.is_number(False)
```
Comment (jreback) [inline pandas/core/dtypes/inference.py]: could also add a comment here instead

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by tapumar** [comment]
Changes done :)

**Review by jorisvandenbossche — APPROVED** [review]
Looks good, thanks!

**Pull request merged**